### PR TITLE
Introduce boxcutter.rb for osx-vm-templates

### PR DESCRIPTION
**Pull request by americanhanko**
## This PR

- Adds a [Special operations](https://en.wikipedia.org/wiki/Special_operations) recipe, designed to be used in conjunction with Tim Sutton's [osx-vm-templates](https://github.com/timsutton/osx-vm-templates). For this recipe to work:

  - It must be used with my fork of [osx-vm-templates](https://github.com/americanhanko/osx-vm-templates/tree/provision-with-chef), which sets Chef as the primary provisioner (amongst a few other things).
  - You must have the `berks` gem installed - this is run `berks vendor`, which passes the cookbooks to the virtual machine during the first boot.

## Notes
- The hope is that this recipe will be able to do everything that `veewee-config.pkg` currently does, eliminating the need to install an unsigned package, making `prepare_iso.sh` the preferred method of creating a macOS Vagrant box with Packer.

### CODE REVIEW

File: attributes/default.rb [hunk 1]
```diff
@@ -11,5 +11,27 @@
 default['macos']['remote_login_enabled'] = true
 default['macos']['disk_sleep_disabled'] = false
 
-default['macos']['network_time_server'] = 'time.windows.com'
+default['macos']['network_time_server'] = 'time.apple.com'
 default['macos']['time_zone'] = 'America/Los_Angeles'
+
+node.default['authorization']['sudo']['groups'] = 'wheel'
```
Comment (jazaval) [inline attributes/default.rb]: `node` is always implied in this file - is there a specific need for this prefix?

File: recipes/boxcutter.rb [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+plist '/private/var/db/com.apple.xpc.launchd/disabled.plist' do
+  entry 'com.apple.screensharing'
+  value false
+  only_if { current_version >= yosemite }
+end
+
+plist '/private/var/db/launchd.db/com.apple.launchd/overrides.plist' do
+  entry 'com.openssh.sshd'
+  value false
+  only_if { current_version < yosemite }
```
Comment (jazaval) [inline recipes/boxcutter.rb]: Does this whole recipe offer support for Mavericks and earlier?

File: spec/unit/recipes/boxcutter_spec.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+require 'spec_helper'
+
+describe 'macos::boxcutter' do
+  context 'macOS High Sierra 10.13' do
```
Comment (jazaval) [inline spec/unit/recipes/boxcutter_spec.rb]: These don't seem to pass for me:

```
1) macos::boxcutter macOS High Sierra 10.13 behaves like a successful convergence does not raise an exception
     Failure/Error: expect { chef_run }.to_not raise_error

       expected no Exception, got #<Chef::Exceptions::ValidationFailed: Property value must be one of: TrueClass, FalseClass, String, Integer, Float!  You passed [nil].> with backtrace:
         # /var/folders/x5/6v905wsn4t5_crp4twz0d0w80000gp/T/d20180226-70017-xetk8p/cookbooks/macos/recipes/boxcutter.rb:29:in `block in from_file'
         # /var/folders/x5/6v905wsn4t5_crp4twz0d0w80000gp/T/d20180226-70017-xetk8p/cookbooks/macos/recipes/boxcutter.rb:27:in `from_file'
         # ./spec/spec_helper.rb:28:in `block (2 levels) in <top (required)>'
         # ./spec/spec_helper.rb:33:in `block (4 levels) in <top (required)>'
         # ./spec/spec_helper.rb:33:in `block (3 levels) in <top (required)>'
     Shared Example Group: "a successful convergence" called from ./spec/unit/recipes/boxcutter_spec.rb:8
     # ./spec/spec_helper.rb:33:in `block (3 levels) in <top (required)>'
```
Looks like `node['macos']['uid']` is `nil` - how is this set?

**Comment by americanhanko** [comment]
Closing this. Given the changes required for the `plist` resource, it should be separated into two PRs: one for the `plist` resource refactor and the other for the `boxcutter.rb` recipe.

**Pull request closed without merging**